src/main/java/net/rakugakibox/springbootext/logback/access/tomcat/RequestAttributeRegardingAccessEvent.java

```diff
@@ -0,0 +1,91 @@
+package net.rakugakibox.springbootext.logback.access.tomcat;
+
+import ch.qos.logback.access.spi.AccessEvent;
+import ch.qos.logback.access.spi.ServerAdapter;
+import org.apache.catalina.AccessLog;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * AccessEvent that regards requestAttributes that may be set by other valves.
+ */
+public class RequestAttributeRegardingAccessEvent extends AccessEvent {
+
+    private String remoteAddress;
+    private String remoteHost;
+    private String protocol;
+    private Integer localPort;
+
+    public RequestAttributeRegardingAccessEvent(HttpServletRequest httpRequest, HttpServletResponse httpResponse,
+                                                ServerAdapter adapter) {
+        super(httpRequest, httpResponse, adapter);
+        prepareForDeferredProcessing();
```

Comment:
I'll have to look into that again. I think it is a left-over from testing where i had some problems with an asynchronous logger (logback-to-logstash). Console did work fine anyway.